src/runtimes/node/bundlers/nft/es_modules.ts

```diff
@@ -81,6 +88,17 @@ export const processESM = async ({
   const entrypointIsESM = isEntrypointESM({ basePath, esmPaths, mainFile })
 
   if (!entrypointIsESM) {
+    if (featureFlags.zisi_functions_api_v2) {
+      throw new FunctionBundlingUserError(
+        `The function '${name}' is written as a commonJS module, but it needs to be written as an ES module.`,
```

Comment:
I think this is a bit simpler and more actionable. We can point the shortlink to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules.

```suggestion
        `The function '${name}' must use the ES module syntax. To learn more, visit https://ntl.fyi/esm.`,
```